integer/src/cmp.rs

```diff
@@ -121,3 +121,32 @@ impl AbsCmp<IBig> for UBig {
         self.0.as_typed().cmp(&rhs.0.as_sign_typed().1)
     }
 }
+
+macro_rules! impl_partial_ord_for_prim_ints {
```

Comment:
For comparison between different types, it's designed to use the `num-order` crate right now. Could you please take a look at that crate and use the `num_cmp` and `num_eq` functions provided in that crate?